core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClientBuilder.java

```diff
@@ -38,43 +37,53 @@
  */
 public abstract class AbstractRetryingClientBuilder<O extends Response> {
 
-    @Nullable
-    private final RetryRule retryRule;
+    @Nullable private final RetryConfigBuilder<O> retryConfig;
 
-    @Nullable
-    private final RetryRuleWithContent<O> retryRuleWithContent;
-
-    private int maxTotalAttempts = Flags.defaultMaxTotalAttempts();
-    private long responseTimeoutMillisForEachAttempt = Flags.defaultResponseTimeoutMillis();
+    private RetryConfigMapping<O> mapping;
 
     /**
      * Creates a new builder with the specified {@link RetryRule}.
      */
     AbstractRetryingClientBuilder(RetryRule retryRule) {
-        this(requireNonNull(retryRule, "retryRule"), null);
+        retryConfig = RetryConfig.builder(requireNonNull(retryRule, "retryRule"));
+        mapping = (ctx, req) -> retryConfig.build();
     }
 
     /**
      * Creates a new builder with the specified {@link RetryRuleWithContent}.
      */
     AbstractRetryingClientBuilder(RetryRuleWithContent<O> retryRuleWithContent) {
-        this(null, requireNonNull(retryRuleWithContent, "retryRuleWithContent"));
+        retryConfig = RetryConfig.builder(requireNonNull(retryRuleWithContent, "retryRuleWithContent"));
+        mapping = (ctx, req) -> retryConfig.build();
+    }
+
+    /**
+     * Creates a new builder with the specified {@link RetryRuleWithContent}.
+     */
+    AbstractRetryingClientBuilder(RetryRuleWithContent<O> retryRuleWithContent, int maxContentLength) {
+        retryConfig = RetryConfig.builder(
+                requireNonNull(retryRuleWithContent, "retryRuleWithContent"),
+                maxContentLength);
```

Comment:
How about removing this constructor as well as `RetryConfig.builder()` that requires `maxContentLength`? We could instead add a setter to the builders?

Reply:
Done